src/js/BuildApi.js

```diff
@@ -1,184 +1,207 @@
 import { gui_log } from "./gui_log";
 import { i18n } from "./localization";
 import { get as getStorage, set as setStorage } from "./SessionStorage";
-import $ from "jquery";
+import CONFIGURATOR from "./data_storage.js";
 
 export default class BuildApi {
     constructor() {
         this._url = "https://build.betaflight.com";
         this._cacheExpirationPeriod = 3600 * 1000;
     }
 
-    load(url, onSuccess, onFailure) {
+    isSuccessCode(code) {
+        return code === 200 || code === 201 || code === 202;
+    }
+
+    async fetchText(url) {
+        const response = await fetch(url, {
+            method: "GET",
+            headers: {
+                "User-Agent": navigator.userAgent,
+                "X-CFG-VER": `${CONFIGURATOR.version}`,
+            },
+        });
+
+        if (this.isSuccessCode(response.status)) {
+            return await response.text();
+        }
+
+        gui_log(i18n.getMessage("buildServerFailure", [url, `HTTP ${response.status}`]));
+        return null;
+    }
+
+    async fetchJson(url) {
+        const response = await fetch(url, {
+            method: "GET",
+            headers: {
+                "User-Agent": navigator.userAgent,
+                "X-CFG-VER": `${CONFIGURATOR.version}`,
+            },
+        });
+
+        if (this.isSuccessCode(response.status)) {
+            return await response.json();
+        }
+
+        gui_log(i18n.getMessage("buildServerFailure", [url, `HTTP ${response.status}`]));
+        return null;
+    }
+
+    async fetchCachedJson(url) {
         const dataTag = `${url}_Data`;
         const cacheLastUpdateTag = `${url}_LastUpdate`;
 
-        const result = getStorage([cacheLastUpdateTag, dataTag]);
-        const dataTimestamp = $.now();
-        const cachedData = result[dataTag];
-        const cachedLastUpdate = result[cacheLastUpdateTag];
+        const storageResult = getStorage([cacheLastUpdateTag, dataTag]);
+        const dataTimestamp = Date.now();
+        const cachedData = storageResult[dataTag];
+        const cachedLastUpdate = storageResult[cacheLastUpdateTag];
 
-        const cachedCallback = () => {
+        if (cachedData && cachedLastUpdate && dataTimestamp - cachedLastUpdate < this._cacheExpirationPeriod) {
             if (cachedData) {
```

Comment:
This will already be true

Reply:
ah of course ... forgot to remove the existing line